src/test/java/com/inventorsoft/junit/service/BookServiceTest.java

```diff
@@ -0,0 +1,230 @@
+
+        book.setId(ID);
+        book.setTitle(TITLE);
+        book.setReleaseDate(RELEASE_DATE);
+        book.setDescription(DESCRIPTION);
+        book.setAuthor(author);
+
+        bookResponse.setId(ID);
+        bookResponse.setTitle(TITLE);
+        bookResponse.setReleaseDate(RELEASE_DATE);
+        bookResponse.setDescription(DESCRIPTION);
+        bookResponse.setAuthor(authorResponse);
+    }
+
+    @Test
+    void getById() {
+        // given
+        when(bookRepository.findById(ID)).thenReturn(Optional.of(book));
+        when(bookMapper.mapEntityToResponse(book)).thenReturn(bookResponse);
+
+        // when
+        BookResponse givenResult = bookService.getById(ID);
+
+        // then
+        assertEquals(bookResponse, givenResult);
+    }
+
+    @Test
+    void getByIdShouldThrowRuntimeExceptionWhenUserNotFound() {
+        assertThrows(RuntimeException.class, () -> bookService.getById(null));
+    }
+
+    @Test
+    void getAll() {
+        // given
+        List<BookResponse> bookResponses = List.of(bookResponse);
+        List<Book> books = List.of(book);
+
+        when(bookMapper.mapEntitiesToResponses(books)).thenReturn(bookResponses);
+        when(bookRepository.findAll()).thenReturn(books);
+
+        // when
+        List<BookResponse> result = bookService.getAll();
+
+        // then
+        assertEquals(bookResponses.size(), result.size());
+        assertArrayEquals(bookResponses.toArray(), result.toArray());
+    }
+
+    @Test
+    void create() {
+        // given
+        CreateBookRequest createBookRequest = new CreateBookRequest();
+        createBookRequest.setAuthorId(ID);
+        createBookRequest.setTitle(TITLE);
+        createBookRequest.setDescription(DESCRIPTION);
+        createBookRequest.setReleaseDate(RELEASE_DATE);
+
+        when(bookMapper.mapCreateRequestToEntity(createBookRequest)).thenReturn(book);
+        when(bookRepository.save(book)).thenReturn(book);
+
+        // when
+        Long bookId = bookService.create(createBookRequest);
+
+        // then
+        assertEquals(ID, bookId);
+        verify(bookRepository, times(1)).save(book);
+        verify(bookMapper, times(1)).mapCreateRequestToEntity(createBookRequest);
+    }
+
+    @Test
+    void update() {
+        // given
+        UpdateBookRequest updateBookRequest = new UpdateBookRequest();
+        updateBookRequest.setAuthorId(ID);
+        updateBookRequest.setTitle(TITLE);
+        updateBookRequest.setDescription(DESCRIPTION);
+        updateBookRequest.setReleaseDate(RELEASE_DATE);
+
+        Book updateBook = new Book();
+        updateBook.setId(book.getId());
+        updateBook.setTitle(book.getTitle());
+        updateBook.setReleaseDate(book.getReleaseDate());
+        updateBook.setDescription(book.getDescription());
+
+        when(bookRepository.findById(ID)).thenReturn(Optional.of(book));
+        bookMapper.updateEntityFromUpdateRequest(book, updateBookRequest);
```

Comment:
this method is not mocked here, it is basically just called, you basically can remove it from here and use verify on it later